### PR TITLE
Rework and simplify update checking logic

### LINKED ISSUE #92593: Project manager lists unavailiable update for "4.3-4.3"
### Tested versions

happens in Godot_v4.3-beta1
not happens in Godot_v4.3-dev6

### System information

Windows 11 - Godot_v4.3-beta1

### Issue description

when I open Godot, the bottom of the Project Manager is written: "Update available: 4.3-4.3.".
When I click on it it brings me to https://godotengine.org/download/archive/4.3-4.3 where I encounter a 404 error.

### Steps to reproduce

open Godot_v4.3-beta1_win64.exe after unzip.

### Minimal reproduction project (MRP)

there is no need for this because the bug is not related to any project

### PR DESCRIPTION
**Pull request by KoBeWi**
Fixes #92593

I have rewritten the update checking logic based on the new format from https://github.com/godotengine/godot-website/pull/858. Unfortunately there was a nasty bug in the previous version that will make users not get notified about beta2 .-.

- The main simplification is that `releases` array is no longer iterated; instead only the first (newest) release is checked.
- Removed `UpdateStatus::DEV`, which was effectively unused and made logic more complex.
- Changed `[]` to `get()`, so it's safer now (it didn't crash, but it's better to be safe).

Some more testing appreciated. For testing different versions faster, you can replace the code that loads JSON with a custom one that loads a local file: `String s = FileAccess::get_file_as_string("somewhere/versions.json");`.

### CODE REVIEW

File: editor/engine_update_label.h [hunk 1]
```diff
@@ -52,15 +52,14 @@ class EngineUpdateLabel : public LinkButton {
 	enum class VersionType {
 		STABLE,
 		RC,
-		BETA,
 		ALPHA,
+		BETA,
```
Comment (AThousandShips) [inline editor/engine_update_label.h:56]: Why this change? This would consider an alpha version as newer than a beta one

Comment (KoBeWi) [reply editor/engine_update_label.h:56]: 🤦‍♂️

**Comment by SpockBauru** [comment]
Got the windows artifact. Now it shows nothing so... It's working?

**Comment by SpockBauru** [comment]
Tested the windows artifact, now it doesn't show anything so... It's working?
How to do the json thing?

**Comment by KoBeWi** [comment]
If nothing appears it means there is no new version available.
You can get the official JSON here: https://godotengine.org/versions.json
However testing custom versions requires compiling a custom build.

**Comment by SpockBauru** [comment]
Got the branch on my repo (yes, GitHub Desktop works!), compiled, changed `VERSION_STATUS` to `"dev6"` in `version_generated.gen.h`, compiled again and now it seems to work :D
The link correctly opens `https://godotengine.org/download/archive/4.3-beta1/`

image

**Review by akien-mga — APPROVED** [review]
Code looks good, I'll try to do some tests to make sure it works for different build names.

**Comment by akien-mga** [comment]
Did some quick testing by adding 4.3-beta2, 4.3-stable, and 4.4-dev1, and making a local build named 4.3-beta1. Seems to work as expected.

**Comment by akien-mga** [comment]
Thanks!
